modules/nw-egressnetworkpolicy-about.adoc

```diff
@@ -0,0 +1,72 @@
+
+[IMPORTANT]
+====
+You must have OpenShift SDN configured to use either the network policy or multitenant modes to configure egress firewall policy.
+
+If you are using network policy mode, egress policy is compatible with only one policy per namespace, and will not work with projects that share a network, such as global projects.
+====
+
+[CAUTION]
+====
+Egress network policy rules do not apply to service endpoint traffic. An Ingress Controller handles traffic for a service, bypassing the egress firewall.
+====
+
+[id="limitations-of-an-egress-firewall"]
+== Limitations of an egress firewall
+
+An egress firewall has the following limitations:
+
+* No project can have more than one EgressNetworkPolicy object.
+
+* The `default` project cannot use egress network policy.
+
+* When using the OpenShift SDN network provider in multitenant mode the following limitations apply:
+
+  - Any project that has been made global via `oc adm pod-network make-projects-global` cannot use egress network policy.
+
+  - Projects merged together via `oc adm pod-network join-projects` cannot use egress network policy in any of the joined projects.
+
+Violating any of these restrictions results in broken egress network policy for the project, and may cause all external network traffic to be dropped.
+
+[id="policy-rule-order"]
+== Matching order for egress network policy rules
+
+The egress network policy rules are evaluated in the order that they are defined, from first to last. The first rule that matches an egress connection from a Pod applies. Any subsequent rules are ignored for that connection.
+
+[id="domain-name-server-resolution"]
+== How Domain Name Server (DNS) resolution works
+
+If you use DNS names in any of your egress network policy rules, proper resolution of the domain names is subject to the following:
+
+* Domain name updates are polled based on the TTL (time to live) value of the domain returned by the local non-authoritative servers.
+
+* The Pod should resolve the domain from the same local name servers when necessary, otherwise the IP addresses for the domain known by the egress network policy controller and the Pod will be different, and the egress network policy may not be enforced as expected.
+
+* The egress network policy controller and Pods asynchronously poll the same local name server. There can be a race condition where the Pod may get the updated IP before the egress controller. Due to this current limitation, domain name usage in EgressNetworkPolicy objects is only recommended for domains with infrequent IP address changes.
+
+[NOTE]
+====
+The egress firewall always allows Pods access to the external interface of the node that the Pod is on for DNS resolution.
+If your DNS resolution is not handled by something on the local node, then you will need to add egress firewall rules allowing access to the DNS server’s IP addresses if you are using domain names in your Pods.
```

Comment:
Can we be more specific than "something?"

s/will need to/must
s/allowing/that allow

Wait, does this only apply if your DNS resolution is not handled by something on the local node and if you are using domain names in your Pods? If so, put both of these restrictions in the first phrase.